Resources/public/js/components/news/form/main.js

```diff
@@ -0,0 +1,101 @@
+
+    return {
+
+        defaults: {
+            templates: {
+                form: form,
+                url: '/admin/api/news<% if (!!id) { %>/<%= id %><% } %>'
+            },
+            translations: {
+                title: 'public.title',
+                content: 'news.content'
+            }
+        },
+
+        header: {
+            title: 'news.headline',
+            toolbar: {
+                buttons: {
+                    save: {
+                        parent: 'saveWithOptions'
+                    }
+                }
+            }
+        },
+
+        layout: {
+            content: {
+                width: 'fixed',
+                leftSpace: true,
+                rightSpace: true
+            }
+        },
+
+        initialize: function() {
+            this.render();
+
+            this.bindDomEvents();
+            this.bindCustomEvents();
+        },
+
+        render: function() {
+            this.$el.html(this.templates.form({translations: this.translations}));
+
+            this.form = this.sandbox.form.create('#news-form');
+            this.form.initialized.then(function() {
+                this.sandbox.form.setData('#news-form', this.data || {});
+            }.bind(this));
+        },
+
+        bindDomEvents: function() {
+            this.$el.find('input, textarea').on('keypress', function() {
+                this.sandbox.emit('sulu.header.toolbar.item.enable', 'save');
+            }.bind(this));
+        },
+
+        bindCustomEvents: function() {
+            this.sandbox.on('sulu.toolbar.save', this.save.bind(this));
+        },
+
+        save: function(action) {
+            if (!this.sandbox.form.validate('#news-form')) {
+                return;
+            }
+
+            var data = this.sandbox.form.getData('#news-form'),
+                url = this.templates.url({id: this.options.id});
+
+            this.sandbox.util.save(url, !this.options.id ? 'POST' : 'PUT', data).then(function(response) {
+                this.afterSave(response, action);
+            }.bind(this));
+        },
+
+        afterSave: function(response, action) {
+            this.sandbox.emit('sulu.header.toolbar.item.disable', 'save');
+
+            if (action === 'back') {
+                this.sandbox.emit('sulu.router.navigate', 'example/news');
+            } else if (action === 'new') {
+                this.sandbox.emit('sulu.router.navigate', 'example/news/add');
+            } else if (!this.options.id) {
+                this.sandbox.emit('sulu.router.navigate', 'example/news/edit:' + response.id);
+            }
+        },
+
+        loadComponentData: function() {
+            var promise = $.Deferred();
+
+            if (!this.options.id) {
+                promise.resolve();
+
+                return promise;
+            }
+            this.sandbox.util.load(_.template(this.defaults.templates.url, {id: this.options.id})).done(function(data) {
```

Comment:
Yups. The problem still valid :(

On Tue, May 24, 2016, 7:03 PM Johannes Wachter <notifications@github.com>
wrote:

> In Resources/public/js/components/news/form/main.js
> <https://github.com/sulu/ExampleNewsBundle/pull/4#discussion_r64370657>:
>
> > +            } else if (action === 'new') {
> > +                this.sandbox.emit('sulu.router.navigate', 'example/news/add');
> > +            } else if (!this.options.id) {
> > +                this.sandbox.emit('sulu.router.navigate', 'example/news/edit:' + response.id);
> > +            }
> > +        },
> > +
> > +        loadComponentData: function() {
> > +            var promise = $.Deferred();
> > +
> > +            if (!this.options.id) {
> > +                promise.resolve();
> > +
> > +                return promise;
> > +            }
> > +            this.sandbox.util.load(_.template(this.defaults.templates.url, {id: this.options.id})).done(function(data) {
>
> sorry i have forgotten this problem (: is this still a valid problem?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/sulu/ExampleNewsBundle/pull/4/files/737dc1f542467caa9ada59b4232bae8a85d06fe9#r64370657>
>
-- 
Budi Arsana
Mobile: +62 817623397